File: string/14_longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
 const longestCommonPrefix = (strs) => {
    let prefix = strs[0];
    for (let i = 0; i < strs.length; i++) {
        tmp = "";
        for (let j = 0; j < strs[i].length; j++) {
            if (strs[i][j] === prefix[j]) {
                tmp += prefix[j];
            }
            else {
                break;
            }
        }
        prefix = tmp;
    }
    return prefix;
};

File: leetcode/py/14_longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        pre = strs[0]
        strs_len = len(strs)
        for i in range(1, strs_len):
            tmp = ""
            for j in range(0, min(len(strs[i]), len(pre))):
                if strs[i][j] == pre[j]:
                    tmp += pre[j]
                else:
                    break
            pre = tmp
        return pre
